src/main/java/org/jabref/logic/layout/format/CffDate.java

```diff
@@ -0,0 +1,62 @@
+package org.jabref.logic.layout.format;
+
+import java.time.LocalDate;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import org.jabref.logic.layout.LayoutFormatter;
+
+public class CffDate implements LayoutFormatter {
+
+    /*
+        This class is used to parse dates for CFF exports. Since we do not know if the input String contains
+        year, month and day, we must go through all these cases to return the best CFF format possible.
+        Different cases are stated below.
+
+        Year, Month and Day contained   => date-released: yyyy-mm-dd
+        Year and Month contained        => month: mm
+                                           year: yyyy
+        Year contained                  => year: yyyy
+        Poorly formatted                => issue-date: <fieldText>
+    */
```

Comment:
Please make a class comment -> JAvaDoc. Put it above line 11 and with `/** ... */`.